### DISCUSSION TITLE
[Markdown] Footnotes are not displayed correctly

### DISCUSSION THREAD

**Discussion by youngfish42** [community/community#21005] [Projects and Issues]
I am writing a readme.md file in my own **[repository](https://github.com/youngfish42/Awesome-Federated-Learning-on-Graph-and-Tabular-Data)** using the Markdown language, but I find that the footnotes are not displayed correctly

As you can see from this page, although the location of the marked footnote generates a hyperlink, there seems to be a problem with the hyperlink numbering between the footnote and the content of the footnote, causing them not to jump correctly to each other.

![图片](https://user-images.githubusercontent.com/38998412/180216721-2ea96247-5cb2-44de-af9c-0f4b68f55d51.png)
https://github.com/youngfish42/Awesome-Federated-Learning-on-Graph-and-Tabular-Data#user-content-fn-FedWalk-779e60a63eb5f43ff1fbe60dac46e6fc

![图片](https://user-images.githubusercontent.com/38998412/180216922-083e277b-a07b-4e22-8dce-a9c18a3b3a57.png)
https://github.com/youngfish42/Awesome-Federated-Learning-on-Graph-and-Tabular-Data#user-content-fnref-FedWalk-779e60a63eb5f43ff1fbe60dac46e6fc

The hyperlinks generated by the footnotes appear as fields like `#user-content-fnref-FedWalk-779e60a63eb5f43ff1fbe60dac46e6fc`
 fields, but they **don't help the hyperlink find the correct location** in the project repository.

I have checked GFM's Syntax [guide](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#footnotes) on footnotes, and related [discussions](https://stackoverflow.com/questions/25579868/how-to-add-footnotes-to-github-flavoured-markdown/), but still can't solve the problem.

**Comment by radar-blip** [comment]
Footnotes have to be in the correct order from 1 to 9 (or however many you have). Also, you have doubled some footnotes at the footer which isnt correct. You must check your markup. 

1 reference number for every footnote (beginning with 1).

I have an example at the bottom of this repo that might help: https://github.com/linuxlawson/gitmark

Hope this helps.

**Reply by youngfish42** [reply]
Thank you for your reply. 😄 

Do you mean that footnotes must be marked with numbers? But I noticed that in the official tutorial you can use English characters as markers.
(see the example of using `note` as marker at the end of the [official tutorial](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#footnotes))

**Reply by radar-blip** [reply]
I have never seen the ```note``` example. I dont know how you would put that in any kind of order.

**Reply by youngfish42** [reply]
![图片](https://user-images.githubusercontent.com/38998412/180358514-f7514b02-a46f-4383-9191-f89a5d374260.png)

Please refer to the section marked in orange

**Reply by radar-blip** [reply]
Yes I do see that now. I looked at it.  👍🏼

**Comment by radar-blip** [comment]
I was only using numbers as an example (sorry for any confusion). I guess you could use letters or other symbols but they must match up with the footnotes.

**Reply by youngfish42** [reply]
Thank you for your generous help!
In my repository, I did set a unique identification mark for each footnote (you can check the source code of my repository if it is convenient)

I can jump between footnotes and quoted content correctly locally using a Markdown editor like Typora, but I'm having problems viewing it online on Github. That's why I opened this discussion.

**Comment by radar-blip** [comment]
In your footnotes (in footer of page) you have used the reference number 2 twice. I can see it in the first two footnotes. I would start by straightening that out.

**Reply by youngfish42** [reply]
LOL I'm also interested in this issue (I guess another discussion could be started).
However, I find that if the footnote is used only once, the above problem still exists

**Reply by radar-blip** [reply]
Are you sure your reference numbers are matching up with your footnotes (in the markup) ??
And only used once?

**Reply by radar-blip** [reply]
You're using words (or parts of words) in your footnote code. You need numbers lol

**Reply by radar-blip** [reply]
**Ignore this, this is incorrect**
~~I think I see whats going on here. You're reference numbers (or words) need to be writtin in superscript. <sup>likethis</sup>~~ Then make sure the footnote matches.
I think a numbered system would be best for as many footnotes as you have.

**Reply by youngfish42** [reply]
You can search and view the `FLDetector` example, this footnote is mentioned and used **only once**.

Here is what it looks like in Github:
![图片](https://user-images.githubusercontent.com/38998412/180362436-26299491-325d-4a52-8cc7-654c3253c747.png)

![图片](https://user-images.githubusercontent.com/38998412/180362481-85d5cf34-b4cd-46a0-af63-957e376dc818.png)



The following is the source code of the corresponding part:
![图片](https://user-images.githubusercontent.com/38998412/180362751-fe129507-b308-4b38-954e-44b7e6fc340a.png)

![图片](https://user-images.githubusercontent.com/38998412/180362778-32aea469-b94c-4239-bb59-1e6ff0bf1696.png)





And finally, here's how it looks in Typora:
![图片](https://user-images.githubusercontent.com/38998412/180362647-5f42ad09-3dd9-4ef0-8c46-3447a9f926b9.png)

**Reply by youngfish42** [reply]
> https://github.com/github-community/community/discussions/21005#discussioncomment-3203108

Thanks for your reply, here I am more curious about the official use of string based footnotes (the case of [note](https://github.com/github-community/community/discussions/21005#discussioncomment-3202959)) but why it doesn't jump correctly in my repository.

**Reply by radar-blip** [reply]
You are right about that one. I dont know what is going on there.
